src/python/WMCore/WorkQueue/DataStructs/WorkQueueElementsSummary.py

```diff
@@ -3,10 +3,88 @@
 
 """
 from __future__ import (print_function, division)
-from pprint import pprint
 from collections import defaultdict
+
 from WMCore.WorkQueue.DataStructs.WorkQueueElementResult import WorkQueueElementResult
 
+
+def getGlobalSiteStatusSummary(elements, status=None, dataLocality=False):
+    """
+    _getGlobalSiteStatusSummary_
+
+    Given a dict with element details keyed by status, such as this format:
+    {u'Available': [{u'Inputs': {}, u'Jobs': 18,...}, {u'Jobs': 11,...}],
+     u'Acquired': [{'Priority': 190000,..}, ...]}
+
+    Creates a summary of jobs and number of wq elements in each status
+    distributed among the sites whitelisted. There are 2 job distribution:
+     *) unique top level jobs per site and per status and (equally
+        distributed among all sites)
+     *) possible top level jobs per site and per status (consider all
+        jobs can run in a single location)
+
+    If status is provided, then skip any workqueue element not in the
+    given status. Otherwise filter only active workqueue status.
+
+    If dataLocality is set to True, then it considers only sites that pass
+    the data location constraint.
+    """
+    if status and isinstance(status, basestring):
+        activeStatus = [status]
+    elif status and isinstance(status, (list, tuple)):
+        activeStatus = status
+    else:
+        activeStatus = ['Available', 'Negotiating', 'Acquired']
+
+    uniqueJobsSummary = {}
+    possibleJobsSummary = {}
+
+    for st in activeStatus:
+        uniqueJobsSummary.setdefault(st, {})
+        possibleJobsSummary.setdefault(st, {})
+        for elem in elements.get(st, []):
+            if dataLocality:
+                possibleSites = getPossibleSites(elem)
+            else:
+                possibleSites = list(set(elem['SiteWhitelist']) - set(elem['SiteBlacklist']))
+
+            try:
+                jobsPerSite = elem['Jobs'] / len(possibleSites)
+            except ZeroDivisionError:
+                possibleSites = ['NoPossibleSite']
+                jobsPerSite = elem['Jobs']
```

Comment:
We don't really know if it's intentional or not, that's why I assigned this WQE to NoPossibleSite, at least we have some idea of how many elements and top level jobs have no common location. Of course, later on we should report on these WQE somehow.